src/runtime_src/core/edge/drm/zocl/sched_exec.c

```diff
@@ -2219,29 +2219,29 @@ static void
 penguin_query(struct sched_cmd *cmd)
 {
 	u32 opc = opcode(cmd);
-	bool cmd_complete = false;
 
 	SCHED_DEBUG("-> penguin_queury() slot_idx=%d\n", cmd->slot_idx);
 	switch (opc) {
 	case ERT_START_COPYBO:
-		if (dma_done(cmd))
-			cmd_complete = true;
+		if (dma_done(cmd)) {
```

Comment:
I suggest a little wrapper on your dam_* functions. Since we have a wrapper to handle dma_handle.dma_flags, it would be neater to treat dma_flags as a private field.
dma_done (just check done bit but not clear it)
dma_error (just check error bit but not clear it)
dma_clear (clear a specific flag ro all flags)